api/src/main/java/com/github/eyefloaters/console/api/service/TopicService.java

```diff
@@ -265,6 +270,7 @@ CompletionStage<Map<Uuid, Either<Topic, Throwable>>> describeTopics(
 
         CompletableFuture.allOf(pendingDescribes)
                 .thenCompose(nothing -> listOffsets(adminClient, result, offsetSpec))
+                .thenCompose(nothing -> describeLogDirs(adminClient, result))
                 .thenApply(nothing -> promise.complete(result))
                 .exceptionally(promise::completeExceptionally);
```

Comment:
What are we getting out of these two calls? The final `thenCompose()` already returns a `CompletableFuture`, so these two calls just seem to be completing `promise` with the same outcome as the one returned by `thenCompose()`.

Reply:
The `thenCompose` returns are `CompletableFuture<Void>` whereas the method here returns `CompletableFuture<Map<...>>`. I find it more readable for the meaningful calls to "align" then finish up separately. Each one is enriching the `result` map, returning nothing, followed by the completion calls.

Aside - `listOffsets` and `describeLogDirs` could probably be called in parallel at some point.

current:
```java
CompletableFuture.allOf(pendingDescribes)
    .thenCompose(nothing -> listOffsets(adminClient, result, offsetSpec))
    .thenCompose(nothing -> describeLogDirs(adminClient, result))
    .thenApply(nothing -> promise.complete(result))
    .exceptionally(promise::completeExceptionally);
```

alternate:
```java
CompletableFuture.allOf(pendingDescribes)
    .thenCompose(nothing -> listOffsets(adminClient, result, offsetSpec))
    .thenCompose(nothing -> {
        describeLogDirs(adminClient, result);
        return result;
    });
```